moto/emr/models.py

```diff
@@ -390,6 +444,28 @@ def add_instance_groups(self, cluster_id, instance_groups):
             result_groups.append(group)
         return result_groups
 
+    def add_instances(self, cluster_id, instances, instance_groups):
+        cluster = self.clusters[cluster_id]
+        response = self.ec2_backend.add_instances(
+            EXAMPLE_AMI_ID, instances["instance_count"], "", [], **instances
+        )
+        for instance in response.instances:
+            instance = FakeInstance(
+                cluster_id=cluster_id,
+                instance_role=instances["instance_role"],
+                ec2_instance_id=instance.id,
+                instance_type=instance.instance_type,
```

Comment:
Is there a specific need for this information to be replicated?
Ideally, we just store the ec2_instance_id, and all the EC2-instance properties (instance_type, public_dns, etc) can be retrieved dynamically when calling `list_instances`

Reply:
Pushed the changes.